### PR TITLE
TableFilters: Fixes filtering with field overrides

### LINKED ISSUE #28598: Table filter with override value mapping
I have applied a value override with the value mapping range to text in the table plugin. Two value mapping is applied with text "ok" and "Not Ok" and it works fine in the table. But when I select "ok" in the table filter, only one row is listed even if there are serval rows with "ok".

Before filter "ok is selected":
image
image
After "Ok" is selected:
image

**Environment**:
- Grafana version: Version 7.2.2
- Data source type & version: Influxdb
- OS Grafana is installed on: Windows Azure
- User OS & Browser: Chrome, Edge
- Grafana plugins: Table

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Fixes issue as described in #28598.

**Which issue(s) this PR fixes**:
Fixes #28598

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by hugohaggmark** [comment]
Merge backport PR after 7.3.1 release please 🙏

File: packages/grafana-ui/src/components/Table/utils.ts [hunk 1]
```diff
@@ -143,16 +148,26 @@ export function calculateUniqueFieldValues(rows: any[], field?: Field) {
...
 
   return set;
 }
 
+export function rowToFieldValue(row: any, field?: Field): string {
+  if (!field || !row) {
+    return '';
+  }
+
+  const index = parseInt(row.id, 10);
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Table/utils.ts:163]: there is an index prop on a row here any reason not to use that?

**Review by torkelo — APPROVED** [review]
Looks good

File: packages/grafana-ui/src/components/Table/utils.ts (hunk 1 above)
Comment (hugohaggmark) [reply packages/grafana-ui/src/components/Table/utils.ts:163]: I didn't know that, is index a more correct way to access the __`column index`__?

Comment (dprokop) [reply packages/grafana-ui/src/components/Table/utils.ts:163]: yes, react-table's Row has in index. But, to get the **column index** we usually keep a counter in the fields loop and pass that counter downstream.

**Review by dprokop — APPROVED** [review]
I really like the test coverage here Hugo!

File: packages/grafana-ui/src/components/Table/utils.ts (hunk 1 above)
Comment (hugohaggmark) [reply packages/grafana-ui/src/components/Table/utils.ts]: Thank you for clarifying, I've changed it to index instead.

**Pull request merged**